packages/patternfly-4/react-core/src/components/Chip/Chip.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ChipButton from './ChipButton';
+import ChipText from './ChipText';
+import ChipWrapper from './ChipWrapper';
+import { TimesCircleIcon } from '@patternfly/react-icons';
+
+const Chip = ({ isOverflowed, children, id, className, ...props }) => {
+  return !isOverflowed ? (
```

Comment:
Can we use a different type of conditional rendering here that would lend itself to more than 2 types of chips? That would help with future proofing the component.

Reply:
I am not sure I understand what you mean.

I can do this differently in the following ways:

```js
const chipBody = !isOverflowed ? (
 <ChipText id={id}>{children}</ChipText>
      <ChipButton aria-label="Remove" id={`remove_${id}`} aria-labelledby={`remove_${id} ${id}`}>
        <TimesCircleIcon aria-hidden="true" />
      </ChipButton>
  ) : (
      <ChipButton aria-label={children}>
        <ChipText>{children}</ChipText>
      </ChipButton>
);

return <ChipWrapper isOverflowed={isOverflowed} className={className} {...props}>{chipBody}</ChipWrapper>
```
or maybe:

```js
const { chipType } = this.props;
switch(chipType) {
case "overflow":
  return (
    <ChipWrapper isOverflowed className={className} {...props}>
      <ChipButton aria-label={children}>
        <ChipText>{children}</ChipText>
      </ChipButton>
    </ChipWrapper>
  );
default:
  return (
    <ChipWrapper className={className} {...props}>
      <ChipText id={id}>{children}</ChipText>
      <ChipButton aria-label="Remove" id={`remove_${id}`} aria-labelledby={`remove_${id} ${id}`}>
        <TimesCircleIcon aria-hidden="true" />
      </ChipButton>
    </ChipWrapper>
  );
}
```

Let me know if you have a better or a different idea. Thanks :fish_cake:

Reply:
I'd take a look at the current patterns used within the library. Maybe something like the button component "variant" prop approach?